packages/grid/data-grid/src/tests/pagination.DataGrid.test.tsx

```diff
@@ -67,6 +68,40 @@ describe('<DataGrid /> - Pagination', () => {
       }, 50);
     });
 
+    it('should trigger onPageChange once if page prop is set', ()=> {
+      const onPageChange = spy();
+
+      render(
+        <div style={{ width: 300, height: 300 }}>
+          <DataGrid {...baselineProps} page={1} onPageChange={onPageChange} pageSize={1} />
+        </div>,
+      );
+      expect(onPageChange.calledOnce).to.be.true;
```

Comment:
?

```suggestion
      expect(onPageChange.callCount).to.equal(1);
```

Reply:
Well it does not mean it's the same page, what if the user changes the page and set it back to the first one?

Reply:
> what if the user changes the page and set it back to the first one?

The developer has provided the page prop, the only way the user can change the page is if the developer change the page prop. So in your case, the page doesn't change.